paz/models/detection/frustum_pointnet.py

```diff
@@ -0,0 +1,370 @@
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.layers import Conv2D, Dense
+from tensorflow.keras.layers import Dropout
+from tensorflow.keras.layers import Input
+from tensorflow.keras.layers import MaxPooling2D
+from tensorflow.keras.models import Model
+
+from examples.frustum_pointnet.dataset_utils import g_mean_size_arr
+from paz.optimization.losses.frustumpointnet_loss import FrustumPointNetLoss
+
+
+def ModelOutputToTensor(output, IntermediateOutputs, NUM_HEADING_BIN=12,
+                        NUM_SIZE_CLUSTER=8):
+    """ Parse batch output to separate tensors (added to IntermediateOutputs)
+    Input:
+        output: TF tensor in shape (B,3+2*NUM_HEADING_BIN+4*NUM_SIZE_CLUSTER)
+        IntermediateOutputs: dict
+    Output:
+        IntermediateOutputs: dict
+
+    """
+    batch_size = output.get_shape()[0]
+    center = tf.slice(output, [0, 0], [-1, 3])
+    IntermediateOutputs['center_boxnet'] = center
+
+    heading_scores = tf.slice(output, [0, 3], [-1, NUM_HEADING_BIN])
+    heading_residuals_normalized = tf.slice(output, [0, 3 + NUM_HEADING_BIN],
+                                            [-1, NUM_HEADING_BIN])
+    IntermediateOutputs['heading_scores'] = heading_scores  # BxNUM_HEADING_BIN
+    IntermediateOutputs['heading_residuals_normalized'] = \
+        heading_residuals_normalized  # BxNUM_HEADING_BIN (-1 to 1)
+    IntermediateOutputs['heading_residuals'] = \
+        heading_residuals_normalized * (np.pi / NUM_HEADING_BIN)
+
+    size_scores = tf.slice(output, [0, 3 + NUM_HEADING_BIN * 2],
+                           [-1, NUM_SIZE_CLUSTER])  # BxNUM_SIZE_CLUSTER
+    size_residuals_normalized = tf.slice(output,
+                                         [0,
+                                          3 + NUM_HEADING_BIN * 2 +
+                                          NUM_SIZE_CLUSTER],
+                                         [-1, NUM_SIZE_CLUSTER * 3])
+
+    size_residuals_normalized = tf.reshape(size_residuals_normalized,
+                                           [batch_size, NUM_SIZE_CLUSTER,
+                                            3])  # BxNUM_SIZE_CLUSTERx3
+    IntermediateOutputs['size_scores'] = size_scores
+    IntermediateOutputs['size_residuals_normalized'] = size_residuals_normalized
+    IntermediateOutputs['size_residuals'] = \
+        size_residuals_normalized * tf.expand_dims(
+            tf.constant(g_mean_size_arr,  dtype=tf.float32), 0)
+
+    return IntermediateOutputs
+
+
+def ObjectPointCloudMasking(point_cloud, mask, npoints=512):
```

Comment:
functions should be named starting with a "verb" and it lower-case with underscore as a separation i.e. `mask_point_cloud`